thavalon-server/src/game/state/on_mission.rs

```diff
@@ -118,6 +119,51 @@ impl GameState<OnMission> {
     fn includes_player(&self, player: &str) -> bool {
         self.proposal().players.contains(player)
     }
+
+    // Append messages for lovers to an exisiting list of effects. This will not modify effects if there are no lovers in game,
+    // so this is safe to call in any game.
+    fn add_lover_effects(&self, effects: &mut Vec<Effect>) {
+        let tristan_display_name = self.game.display_name_from_role(Role::Tristan);
+        let tristan_on_mission = match tristan_display_name {
+            Some(name) => Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(name)),
+            None => None,
+        };
+        let iseult_display_name = self.game.display_name_from_role(Role::Iseult);
+        let iseult_on_mission = match iseult_display_name {
+            Some(name) => Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(name)),
+            None => None,
```

Comment:
For these match statements, you might want to look at using `.map_or()`. That function on an option will either return a default you specified if the option is none or use the contained value in a function. You could use `map_or(false, ...`) to perform the  checks you do if the option contains a value or just return false immediately if it's none.

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or

Reply:
That's pretty handy, but I think I want to keep `x_on_mission` as an Option<bool> rather than a bool here, since the possible values are really true/false/x is not in game.

That really helps condense `build_lover_effect` though, ty!